dotnet/src/Connectors/Connectors.AI.OpenAI/ChatCompletionWithData/AzureOpenAIChatCompletionWithDataService.cs

```diff
@@ -66,7 +66,7 @@ public IAsyncEnumerable<StreamingChatMessageContent> GetStreamingChatMessageCont
     public async Task<IReadOnlyList<TextContent>> GetTextContentsAsync(string prompt, PromptExecutionSettings? executionSettings = null, Kernel? kernel = null, CancellationToken cancellationToken = default)
     {
         return (await this.GetChatMessageContentsAsync(prompt, executionSettings, kernel, cancellationToken).ConfigureAwait(false))
-            .Select(chat => new TextContent(chat.Content, chat, chat.ModelId, Encoding.UTF8, chat.Metadata))
+            .Select(chat => new TextContent(chat.Content, chat.ModelId, chat.Content, Encoding.UTF8, chat.Metadata))
```

Comment:
You need to pass the ChatMessage as the innercontent in here otherwise it will break the InvokeAsync<T> when asking for a `ChatMessageCOntent` when the Service supports `IChatCompletion` and `ITextGeneration`.

```suggestion
            .Select(chat => new TextContent(chat.Content, chat.ModelId, chat, Encoding.UTF8, chat.Metadata))
```

Reply:
Proposed a big change to this as I just realized the InvokeASync<T> today doesn't have access to the Connector output, only the `.Text` string.

Reply:
Fixed